src/cstubs/cstubs_structs.ml

```diff
@@ -15,6 +15,8 @@ sig
   val constant : string -> 'a typ -> 'a const
 
   val enum : string -> ?typedef:bool -> ?unexpected:(int64 -> 'a) -> ('a * int64 const) list -> 'a typ
+
+  val interrogated_abstract : name:string -> 'a abstract typ
 end
```

Comment:
It seems that this `module type` definition could be simplified to

```ocaml
module type TYPE = Ctypes.TYPE
```

Would you like to make that change here?  (I could make a separate PR, but it'll cause a merge conflict with this PR.)

Reply:
I can make this change.

Reply:
Perhaps we should use `Ctypes.TYPE` directly, and not alias the name as `Cstubs_structs.TYPE` at all?

Reply:
Yes, that's a good idea.

We could also deprecate `Cstubs.Types.TYPE` and `Cstubs.FOREIGN`:

```ocaml
   [@@deprecated "Cstubs.Types.TYPE is deprecated. Use Ctypes.TYPE instead"]

   [@@deprecated "Cstubs.FOREIGN is deprecated. Use Ctypes.FOREIGN instead"]
```

to (eventually) simplify the interface and avoid unnecessary run-time dependencies on the `Cstubs` module.

Reply:
Right, I'll do that.

I guess we can't remove the `Cstubs_structs.TYPE` alias, as that would be a breaking change. I'll deprecate it for now instead.